packages/common/src/api/library.ts

```diff
@@ -0,0 +1,74 @@
+import type { full } from '@audius/sdk'
+
+import { createApi } from 'audius-query'
+import { UserCollectionMetadata } from 'models/Collection'
+import { Kind } from 'models/Kind'
+import { makeActivity } from 'services/audius-api-client/ResponseAdapter'
+import { APIActivityV2 } from 'services/audius-api-client/types'
+import { reformatCollection } from 'store/cache/collections/utils/reformatCollection'
+import { encodeHashId } from 'utils/hashIds'
+import { removeNullable } from 'utils/typeUtils'
+
+type GetLibraryAlbumsArgs = {
+  userId: number
+  offset: number
+  limit: number
+  query?: string
+  sortMethod?: full.GetUserLibraryAlbumsSortMethodEnum
+  sortDirection?: full.GetUserLibraryAlbumsSortDirectionEnum
+}
+
+export const libraryApi = createApi({
+  reducerPath: 'libraryApi',
+  endpoints: {
+    getLibraryAlbums: {
+      fetch: async (
+        args: GetLibraryAlbumsArgs,
+        { audiusSdk, audiusBackend }
+      ) => {
+        const {
+          userId,
+          offset,
+          limit,
+          query = '',
+          sortMethod = 'added_date',
+          sortDirection = 'desc'
+        } = args
+        const sdk = await audiusSdk()
+        const { data, signature } =
+          await audiusBackend.signDiscoveryNodeRequest()
+        const { data: rawAlbums = [] } =
+          await sdk.full.users.getUserLibraryAlbums({
+            id: encodeHashId(userId),
+            userId: encodeHashId(userId),
+            offset,
+            limit,
+            query,
+            sortMethod,
+            sortDirection,
+            type: 'all',
+            encodedDataMessage: data,
+            encodedDataSignature: signature
+          })
+        const albumsMetadata = rawAlbums
+          .map((r: APIActivityV2) => makeActivity(r))
+          .filter(removeNullable) as UserCollectionMetadata[]
+        const albums = albumsMetadata.map((am) =>
+          reformatCollection({
+            collection: am,
+            audiusBackendInstance: audiusBackend,
+            omitUser: false
+          })
+        )
+        return albums
+      },
+      options: {
+        kind: Kind.COLLECTIONS,
+        schemaKey: 'collections',
+        idListArgKey: 'playlist_id'
```

Comment:
We don't need `idListArgKey` here.

`idListArgKey` is only useful if you have a hook argument containing the ids of the results you expect. It allows cache hits before making the initial fetch.

Example:
`fetchCollections(collection_ids: ID[])`
`idArgKey: 'collection_ids'`
usage: `useFetchCollections({ collection_ids: ['123', '456'] })` now if collections 123 and 456 are found in the cache, we skip the initial fetch.

However, for this case, we don't know upfront what ids we are going to need, so we always have to fetch and construct the entity list based on the response.

Reply:
ahh thanks for the clarification